COMMIT-GUIDELINES.md

```diff
@@ -0,0 +1,161 @@
+
+The changelog can be found within the [CHANGELOG.md](https://github.com/ongov/ontario-design-system/blob/develop/CHANGELOG.md) file, which a running history of all changes made in each release. The changelog version entries look like the following example:
+
+```
+# 5.0.0-alpha.3 (2024-12-17)
+
+### Bug Fixes
+
+- **grid.mixins:** declare $modules before use 1c7791f
+- **utils:** isNumber function validation inverted 37637b6
+- **utils:** update `retrieveEnumKeys` function 325d3af
+```
+
+#### Breaking changes
+
+A breaking change is a change that is significant enough to cause an incompatibility with the current state of the packages. This could be a removed component property, a set of styles that work differently, or a hole host of different changes.
+
+Breaking changes are denoted in the footer of the commit by starting it with `BREAKING CHANGE:` followed by a message about what the breaking change impacts.
+
+It is highly recommended to add a clear reason for the breaking change and what actions need to be taken when upgrading to the newer version. See the example under [Breaking change](#breaking-change) for an idea of how to communicate the change.
+
+It is important to note: Breaking changes can be added to any commit irregardless of the commit prefix, they are not just relegated to `fix` or `feat` commit prefixes as those aren't always relevant when introducing a breaking change.
+
+## Examples
+
+Good commit messages go along way in creating and maintaining a high quality project. They tell the story of how it grew and evolved overtime. They also help when investigating where things happened, should you ever have to go back in time. Commit messages are almost as important as high quality documentation and code comments.
+
+### Base structure
+
+The base structure of a commit message is:
+
+```
+prefix(optional-scope): subject
+
+body
+
+footer
+```
+
+### Commits with prefix and scope
+
+Adding a prefix and scope helps indicate the impact and theme of the commit and what it impacts.
+
+Commits can be simple:
+
+```
+chore(monorepo): add useWorkspaces to lerna.json
+```
+
+Commits can be more detailed:
+
+```
+fix(utils): isNumber function validation inverted
+
+This fix handles validating numbers are numbers and not numbers are `NaN`.
+
+- Add `undefined` and `null` checks to `isNumber` to guarantee they're
+  false.
+- Add unit tests to validate this case via the `date-validation-utils` function `isInvalidYear`.
+- Add unit tests to validate the `isNumber` util function.
+
+JIRA: DS-1234
+```
+
+Commits can have body text span multiple lines as paragraphs using new lines:
+
+```
+fix(utils): update retrieveEnumKeys function
+
+Updates to `isNumeric` invalided the logic this function was using,
```

Comment:
Very minor change here but I believe `isNumeric` should instead be `isNumber` as that is what is referenced above.

Reply:
I think I made a typo in the original commit 😅 I'll update this sample, but I think it is going down in history! [Yup, I did 🙃](https://github.com/ongov/ontario-design-system/commit/325d3affd394474f3ea20fb97c16aa91b22d44e3)!